src/umap/RegionManager.cpp

```diff
@@ -72,30 +67,23 @@ RegionManager::removeRegion( char* region )
 {
   UMAP_LOG(Debug, "region: " << (void*)region);
 
+  std::lock_guard<std::mutex> lock(m_mutex);
```

Comment:
When does the lock get freed?

Reply:
std::lock_guard<> is RAII and the lock will be unlocked when it goes out of scope.

"The class lock_guard is a mutex wrapper that provides a convenient RAII-style mechanism for owning a mutex for the duration of a scoped block."

Reply:
I was pretty sure that was the case, but was double checking. The mutex should be held for the shortest duration. If holding the mutex until the end of the function is what's intended, that code is OK.